guides/plugins/plugins/framework/flow/action-transactions.md

```diff
@@ -0,0 +1,86 @@
+---
+nav:
+  title: Running actions inside transactions
+  position: 30
+
+---
+
+# Action transactions
+
+## Overview
+
+In this guide, you'll learn how to run your action code inside a transaction. This may be important for you if you want to graciously handle rollbacks in certain scenarios. We have implemented various abstractions to ease this process; however, you need to opt in.
+
+For some more background, please see the ADR [Action Transactions](../../../../../resources/references/adr/2024-02-11-transactional-flow-actions).
+
+## Prerequisites
+
+In order to make your action run inside a database transaction, you will need an existing Flow Action. Therefore, you can refer to the [Add Flow Builder Action Guide.](./add-flow-builder-action)
+
+## Run your action inside a transaction
+
+All you have to do is to implement the `Shopware\Core\Content\Flow\Dispatching\TransactionalAction` interface. It does not have any methods to implement.
+
+When your action implements the interface the Flow Dispatcher will wrap your action in a transaction. If an exception is thrown, it will be caught, the transaction will be rolled back, and an error logged.
```

Comment:
```suggestion
When your action implements the interface the Flow Dispatcher will wrap your action in a transaction. If an exception is thrown, it will be caught, the transaction will be rolled back, and an error is logged.
```